packages/server/src/adapters/fastify/fastifyRequestHandler.ts

```diff
@@ -55,18 +57,94 @@ export async function fastifyRequestHandler<
     },
   });
 
-  const { res } = opts;
+  return iteratorToResponse(resultIterator, opts.res);
+}
+
+/**
+ * @internal
+ */
+export async function iteratorToResponse(
+  iterator: AsyncGenerator<
+    ResponseChunk | HTTPResponse,
+    ResponseChunk | undefined
+  >,
+  res: FastifyReply,
+): Promise<FastifyReply> {
+  const { value: responseInit, done: invalidInit } = await (
+    iterator as AsyncGenerator<HTTPResponse, HTTPResponse | undefined>
+  ).next();
+  const { value: firstChunk, done: abort } = await (
+    iterator as AsyncGenerator<ResponseChunk, ResponseChunk | undefined>
+  ).next();
 
-  if ('status' in result && (!res.statusCode || res.statusCode === 200)) {
-    res.statusCode = result.status;
+  if (invalidInit || (abort && !firstChunk)) {
+    res.statusCode = 500;
+    return res.send();
   }
-  for (const [key, value] of Object.entries(result.headers ?? {})) {
+  if ('status' in responseInit && (!res.statusCode || res.statusCode === 200)) {
+    res.statusCode = responseInit.status;
+  }
+  for (const [key, value] of Object.entries(responseInit.headers ?? {})) {
     /* istanbul ignore if -- @preserve */
     if (typeof value === 'undefined') {
       continue;
     }
-
     void res.header(key, value);
   }
-  await res.send(result.body);
+
+  // iterator is already exhausted, this means we're not streaming the response
+  if (abort) {
+    if (firstChunk) {
+      // case of a full response
+      return res.send(firstChunk[1]);
+    } else {
+      // case of a method === "HEAD" response
+      return res.send();
+    }
+  }
+
+  // iterator is not exhausted, we can setup the streamed response
+  void res.header('Transfer-Encoding', 'chunked');
+  void res.header(
```

Comment:
Is `void` needed here?

Reply:
Yes, it returns a `FastifyReply` (same as `.send()`) which is a thenable (i.e. promise-esque), and the linter wants promises to be explicitly awaited or voided.